net/libfab/libfab.c

```diff
@@ -1168,38 +923,62 @@ static int libfab_addr_port_verify(struct m0_net_transfer_mc *tm, const char *na
  * else returns error code.
  */
 static int libfab_ep_find(struct m0_net_transfer_mc *tm, const char *name,
-			  struct m0_fab__ep_name *epn,
+			  struct m0_net_ip_params *addr,
+			  struct m0_net_end_point *nep,
 			  struct m0_net_end_point **epp)
 {
-	struct m0_net_end_point  *net;
 	struct m0_fab__ep        *ep;
-	uint64_t                  ep_name_n = 0;
-	char                      ep_str[LIBFAB_ADDR_STRLEN_MAX + 9] = {'\0'};
+	struct m0_fab__active_ep *aep;
+	struct m0_net_ip_addr     net_ip = {};
+	struct m0_fab__tm        *ma;
+	char                     *wc = NULL;
 	int                       rc = 0;
+	bool                      found = false;
 
-	if (epn != NULL)
-		libfab_ep_pton(epn, &ep_name_n);
-
+	if (addr != NULL)
+		net_ip.nia_n = *addr;
 	M0_ASSERT(libfab_tm_is_locked(tm->ntm_xprt_private));
-	net = m0_tl_find(m0_nep, net, &tm->ntm_end_points,
-			 libfab_ep_cmp(libfab_ep(net), name, &ep_name_n));
 
-	if (net == NULL) {
+	if (nep != NULL)
+		found = libfab_ep_find_by_ptr(nep, tm, &ep);
+	else if (name != NULL)
+		found = libfab_ep_find_by_str(name, tm, &ep);
+	else
+		found = libfab_ep_find_by_num(&net_ip, tm, &ep);
```

Comment:
Can we do this so number comparison gets priority over string?

	else if (addr != NULL)
		found = libfab_ep_find_by_num(&net_ip, tm, &ep);
	else
		found = libfab_ep_find_by_str(name, tm, &ep);

Reply:
Done.

Reply:
Here also use the 'likely' macro if needed.

Reply:
Done.